Automattic/newspack-blocks
# Add/embed pdf

### All Submissions:






### Changes proposed in this Pull Request:



Closes #897. Allow embedding documents (PDF, Doc, Excel, and PPT) with the Iframe block.

### How to test the changes in this Pull Request:

1. Add an Iframe block, and upload a PDF file as the iframe source.
2. Notice that the PDF is previewed in the editor.
3. Load the page in the front and notice that the PDF file is embedded.

### Other information:

This uses the Google Doc Viewer to embed the documents which is the same way is used when using the [amp-google-document-embed](https://amp.dev/documentation/components/amp-google-document-embed/) tag. The issue with the Google Doc Viewer is that [1 or 2 times out of 10 it doesn't load in iframes](https://stackoverflow.com/questions/40414039/google-docs-viewer-returning-204-responses-no-longer-working-alternatives). A solution for this in `non-amp` pages is to [keep reloading the iframe until its content is loaded](https://github.com/Automattic/newspack-blocks/pull/949/files#diff-e31b7d811bd799e3d22782e7babd1f230b2c28feb2dc5d14976a91e5046ac056R107-R120) which seems to do the trick. But unfortunately, we can't have on AMP since[`onload` is dismissed from the amp-iframe tag](https://github.com/ampproject/amphtml/blob/main/extensions/amp-iframe/1.0/component.js#L25).

A first alternative would be to use Microsoft Office Viewer but it doesn't support PDFs.

Another alternative is to use something like [pdf.js](https://github.com/mozilla/pdf.js) but I'm not sure how it will behave with AMP.
